### PR TITLE
Adding types for js-worker-search

### PR DESCRIPTION
**Pull request by pbalogh**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/js-worker-search/tslint.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules" : {
+        "npm-naming": [true,{"mode":"code","errors":[["NeedsExportEquals",false]]}]
```
Comment (DanielRosenwasser) [inline types/js-worker-search/tslint.json]: Why is this needed?

Comment (pbalogh) [reply types/js-worker-search/tslint.json]: Good catch! This was from an earlier way of handling exports. Removed!

**Comment by pbalogh** [comment]
I believe this is awaiting reviewer feedback, but can't be sure...

File: types/js-worker-search/tslint.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
+        "npm-naming": [
```
Comment (uniqueiniquity) [inline types/js-worker-search/tslint.json]: It looks like this wasn't actually removed as you intended. Could you please remove it? Thanks!

**Review by uniqueiniquity — CHANGES_REQUESTED** [review]

File: types/js-worker-search/tslint.json (hunk 2 above)
Comment (pbalogh) [reply types/js-worker-search/tslint.json:4]: I removed it, but then Travis CI complained that I needed to put it back in -- because of the way that imports are set up in js-worker-search. I can get more info if you need, but it was a frustrating situation where I'd submit one way and get dinged for it, then submit the other way and get dinged for that....

Comment (sandersn) [reply types/js-worker-search/tslint.json:4]: npm-naming is getting confused by the webpack output that js-worker-search ships. So you'll need to keep the rule exemption.

**Comment by uniqueiniquity** [comment]
Congratulations on your first DefinitelyTyped contribution!
Thank you for being a part of the community!

**Review by uniqueiniquity — APPROVED** [review]

**Comment by pbalogh** [comment]
Thank you!
    On Monday, April 6, 2020, 03:57:57 PM EDT, Ben Lichtman <notifications@github.com> wrote:  
 
 


Congratulations on your first DefinitelyTyped contribution!
Thank you for being a part of the community!
